include/linker/common-rom.ld

```diff
@@ -9,7 +9,7 @@
 		 * instruction.  See include/sw_isr_table.h.
 		 */
 		. = ALIGN(CONFIG_ARCH_SW_ISR_TABLE_ALIGN);
-		*(SW_ISR_TABLE)
+		*(_SW_ISR_TABLE_SECTION_NAME)
```

Comment:
That's only defined in one place, why not just use the section name directly?

If the answer is "because it's a .gnu.linkonce section that looks toolchain-dependent", then maybe we should maybe add an approprirate abstraction layer instead of just renaming things.

Reply:
Not limitted to this,  as we are working on adding non-gnu toolchain support. So indeed we need an approprirate abstraction layer for linker script.